azure/templates/run-tests.yml

```diff
@@ -36,6 +36,27 @@ steps:
     displayName: Setup pytests
     condition: always()
 
+  - bash: |
+      proxyURL="https://$APIGEE_ENVIRONMENT.api.service.nhs.uk/$SERVICE_BASE_PATH/comms/_status"
+      proxyResponse=$(
+      curl --request GET $proxyURL \
+           --header 'Content-Type: application/vnd.api+json' \
+           --header 'Authorization: Bearer $bearerToken'
+      )
+      echo "$proxyResponse"
+      errorResponse=$(echo $proxyResponse | jq ".errors[0]")
+      status=$(echo $errorResponse | jq -r ".status")
+      code=$(echo $errorResponse | jq -r ".code")
+
+      if [[ ($status == "404" && $code == "CM_NOT_FOUND") || ($status == "401" && $code == "CM_DENIED") ]]; then
+          exit 0
+      else
+          exit 1
+      fi
+    workingDirectory: $(Pipeline.Workspace)/s/$(SERVICE_NAME)/$(SERVICE_ARTIFACT_NAME)
+    displayName: Check proxy is deployed
+    retryCountOnTaskFailure: 3
```

Comment:
Will this wait between retries, if so how long does it wait?

I think this value may need to be higher to cater for when its taking some time for the proxy to have been stood up.

Reply:
From the azure docs it looks like it has exponential backoff built-in, starting with a few seconds and increasing per retry. They don't give a specific timing value though, I'd expect it varies due to performance.

> The failing task retries in seconds. The wait time between each retry increases after each failed attempt.

Reply:
Excellent, that sounds ideal - I mean not about the azure docs not being clear but the functionality sounds perfect.